packages/mobile/src/lib.rs

```diff
@@ -24,27 +24,56 @@ pub mod launch_bindings {
 }
 
 /// Launch via the binding API
-pub fn launch(incoming: fn() -> Element) {
+pub fn launch(root: fn() -> Element) {
+    launch_cfg(root, vec![], vec![]);
+}
+
+pub fn launch_cfg(
+    root: fn() -> Element,
+    contexts: Vec<Box<dyn Fn() -> Box<dyn Any> + Send + Sync>>,
+    platform_config: Vec<Box<dyn Any>>,
+) {
     #[cfg(target_os = "android")]
     {
-        *APP_FN_PTR.lock().unwrap() = Some(incoming);
+        *APP_OBJECTS.lock().unwrap() = Some(BoundLaunchObjects {
+            root,
+            contexts,
+            platform_config,
+        });
     }
 
     #[cfg(not(target_os = "android"))]
     {
-        dioxus_desktop::launch::launch(incoming, vec![], Default::default());
+        dioxus_desktop::launch::launch(root, contexts, platform_config);
     }
 }
 
-static APP_FN_PTR: Mutex<Option<fn() -> Element>> = Mutex::new(None);
+static APP_OBJECTS: Mutex<Option<BoundLaunchObjects>> = Mutex::new(None);
```

Comment:
Could this be a thread local instead of a static? `Box<dyn Any>` is not send or sync so this looks unsound

Reply:
I need to test on-device to see if the android on_load is guaranteed to be fired from the main thread...

there's a chance it's not and we might need to mess with how this is structured a b it.

Reply:
Okay the way we do `main` has weird semantics. I think technically this is *fine* since the program needs to terminate before the `main()` call can finish up during load. I believe this the same case as an `Rc` moving between threads being sound *only if you can prove it moves completely*. We actually do have an assurance the `Box<dyn Any>` is not held onto anything from the main thread since the main thread no longer exists.

I'll add a note about this in the code.